src/dune_rules/dune_file.mli

```diff
@@ -103,7 +103,7 @@ module Mode_conf : sig
   end
 
   module Set : sig
-    type mode_conf = t
+    type mode_conf := t
```

Comment:
Can you hoist this alias up to share it with `Lib`?

Reply:
done in https://github.com/ocaml/dune/pull/6238/commits/07c6f7095af189fd3fa08194c055b090def6b361.